### PR TITLE
Adding logging support on droid and logging failure message

### PR DESCRIPTION
**Pull request by rohitjain85**
Adding logging support in V8Runtime for droid and logging failure message during exception.

#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and Microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into Microsoft/react-native :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/Microsoft/react-native/pull/17)

### CODE REVIEW

File: ReactCommon/jsi/V8Runtime_shared.cpp [hunk 1]
```diff
@@ -239,6 +239,12 @@ namespace facebook { namespace v8runtime {
     }
   }
 
+  void  V8Runtime::Log(const std::string& message, const unsigned int logLevel) {
+    if (logger_) {
+      (*logger_)("V8Runtime:: " + message, logLevel);
+    }
```
Comment (mganandraj) [inline ReactCommon/jsi/V8Runtime_shared.cpp]: This can be inlined .. just keep the definition along with declaration ..

Comment (rohitjain85) [reply ReactCommon/jsi/V8Runtime_shared.cpp]: Done.

**Comment by rohitjain85** [comment]
@msethims is added to the review. #Closed

File: ReactCommon/jsi/V8Runtime_shared.cpp [hunk 2]
```diff
@@ -283,7 +285,9 @@ namespace facebook { namespace v8runtime {
         sstr << stack_trace_string2 << std::endl;
       }
 
-      throw jsi::JSError(*this, sstr.str());
+      std::string errorMessage{ sstr.str() };
+      Log(errorMessage, 3 /*logLevel*/);
```
Comment (msethims) [inline ReactCommon/jsi/V8Runtime_shared.cpp]: need to comment the category of this magic number

**Pull request closed without merging**